### PR TITLE
Il chembl patterns

### PR DESCRIPTION
**Pull request by ireneisdoomed**
After the revision of the ChEMBL submission, some changes need to be made in the schema:
1. `studyStartDate`. When I added the field I read the [documentation](https://json-schema.org/understanding-json-schema/reference/string.html) for date validation where by specifying that the format is a date, the required YYYY-MM-DD format was implied. This was not the case when validating ChEMBL so I'm adding the pattern. It is very long to specify February having 28 days. We can shorten it to: 
`(\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))`

2. `studyStopReason`. Pattern to avoid any leading or trailing white spaces or newlines

### CODE REVIEW

File: opentargets.json [hunk 1]
```diff
@@ -1323,11 +1323,13 @@
     "studyStartDate": {
       "type": "string",
       "format": "date",
-      "description": "Start date of study in a YYYY-MM-DD format"
```
Comment (DSuveges) [inline opentargets.json:1326]: Shouldn't we just request data field? [JSON Schema allows this type](http://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.1). I'm not 100% sure it applies here though.

File: opentargets.json [hunk 2]
```diff
@@ -1323,11 +1323,13 @@
...
+      "description": "Start date of study in a YYYY-MM-DD format",
+      "pattern": "(((19|20)([2468][048]|[13579][26]|0[48])|2000)[-]02[-]29|((19|20)[0-9]{2}[-](0[469]|11)[-](0[1-9]|[12][0-9]|30)|(19|20)[0-9]{2}[-](0[13578]|1[02])[-](0[1-9]|[12][0-9]|3[01])|(19|20)[0-9]{2}[-]02[-](0[1-9]|1[0-9]|2[0-8])))"
     },
     "studyStopReason": {
       "type": "string",
-      "description": "Reason why a study has been stopped"
+      "description": "Reason why a study has been stopped",
+      "pattern": "^[^\\ ].*[^\\ ]$"
```
Comment (DSuveges) [inline opentargets.json:1332]: Why is this pattern required?

**Review by DSuveges — CHANGES_REQUESTED** [review]

**Comment by DSuveges** [comment]
Are any of these changes require updates in [#1413](https://github.com/opentargets/platform/issues/1413)?

File: opentargets.json (hunk 2 above)
Comment (ireneisdoomed) [reply opentargets.json:1332]: To avoid any leading or trailing white spaces or newlines in this field.

Comment (DSuveges) [reply opentargets.json:1332]: Ah, yes, these weird whitespaces.

**Review by DSuveges — APPROVED** [review]
All looks good.

**Pull request merged**